internal/manifest/v1beta1/client.go

```diff
@@ -54,3 +61,36 @@ func (cc *ClusterClient) GetRESTConfig(
 	}
 	return restConfig, err
 }
+
+func WithClientCacheKey() declarative.WithClientCacheKeyOption {
+	cacheKey := func(ctx context.Context, resource declarative.Object) (any, bool) {
+		logger := log.FromContext(ctx)
+
+		if resource == nil {
+			return nil, false
+		}
+		manifest := resource.(*manifestv1beta1.Manifest)
+		labelValue, err := internal.GetResourceLabel(resource, manifestv1beta1.KymaName)
+		objectKey := client.ObjectKeyFromObject(resource)
+		var labelErr *types.LabelNotFoundError
+		if errors.As(err, &labelErr) {
+			return nil, false
+		}
+		cacheKey := generateCacheKey(labelValue, strconv.FormatBool(manifest.Spec.Remote), manifest.GetNamespace())
+		logger.V(internal.DebugLogLevel).Info(
+			"resource will be cached",
+			"resource", objectKey,
+			"cachekey", cacheKey,
+		)
+		return cacheKey, true
+	}
+	return declarative.WithClientCacheKeyOption{ClientCacheKeyFn: cacheKey}
+}
+
+func generateCacheKey(values ...string) string {
```

Comment:
maybe use `strings.Join()` instead of a custom string builder

Reply:
I use `strings.Join()`, it's a bit flexible if later needs to introduce more values as the source of cache key, although I believe will not happens.